contract_manager/src/contracts/sui.ts

```diff
@@ -185,18 +173,56 @@ export class SuiContract extends Contract {
   async executeMigrateInstruction(vaa: Buffer, keypair: Ed25519Keypair) {
     const tx = new TransactionBlock();
     const packageId = await this.getPythPackageId();
-    let decreeReceipt = await this.getVaaDecreeReceipt(tx, packageId, vaa);
+    let verificationReceipt = await this.getVaaVerificationReceipt(
+      tx,
+      packageId,
+      vaa
+    );
 
     tx.moveCall({
       target: `${packageId}::migrate::migrate`,
-      arguments: [tx.object(this.stateId), decreeReceipt],
+      arguments: [tx.object(this.stateId), verificationReceipt],
     });
 
     return this.executeTransaction(tx, keypair);
   }
 
   async executeUpdatePriceFeed(senderPrivateKey: string, vaas: Buffer[]) {
-    throw new Error("Not implemented");
+    throw new Error("Use executeUpdatePriceFeedWithFeeds instead");
+  }
+
+  getSdkClient(): SuiPythClient {
+    return new SuiPythClient(
+      this.getProvider(),
+      this.stateId,
+      this.wormholeStateId
+    );
+  }
+
+  async executeUpdatePriceFeedWithFeeds(
+    senderPrivateKey: string,
+    vaas: Buffer[],
+    feedIds: string[]
+  ) {
+    const tx = new TransactionBlock();
+    const client = this.getSdkClient();
+    await client.updatePriceFeeds(tx, vaas, feedIds);
+    const keypair = Ed25519Keypair.fromSecretKey(
+      Buffer.from(senderPrivateKey, "hex")
+    );
+    let result = await this.executeTransaction(tx, keypair);
+    return result.digest;
+  }
+  async executeCreatePriceFeed(senderPrivateKey: string, vaas: Buffer[]) {
+    const tx = new TransactionBlock();
+    const client = this.getSdkClient();
+    await client.createPriceFeed(tx, vaas);
```

Comment:
I think I missed this on the previous review but `createPriceFeeds` seem to be a better name.

Reply:
oh, I had this implicit convention that functions starting with `execute` actually do something on-chain. That's why I named it like this